cmd/commands/git-source.go

```diff
@@ -524,6 +524,10 @@ func RunGitSourceList(ctx context.Context, runtimeName string, includeInternal b
 			continue
 		}
 
+		if gs.Self == nil {
```

Comment:
This means you will not show it at all, I wonder if this is what we want. How does UI deal with this state?
We need to have the same behavior

Reply:
currently, in this situation (no self) the ui currently won't show it as well (with the recently-added descriptive error )

Reply:
Oh, but it does show an error, maybe we should also print some error